client/my-sites/site-settings/test/form-general.jsx

```diff
@@ -88,4 +131,84 @@ describe( 'SiteSettingsFormGeneral ', () => {
 			expect( comp.find( 'Banner' ).length ).toBe( 0 );
 		} );
 	} );
+
+	describe( 'Privacy picker - simple sites', () => {
+		let testProps;
+
+		beforeEach( () => {
+			testProps = {
+				...props,
+				siteIsJetpack: false,
+				site: { plan: PLAN_PERSONAL },
+				fields: {
+					blog_public: 1,
+					wpcom_coming_soon: 0,
+				},
+				withComingSoonOption: true,
+				updateFields: fields => {
+					testProps.fields = fields;
+				},
+			};
+		} );
+
+		test( `Should have 4 visibility options`, () => {
+			const { container } = render( <SiteSettingsFormGeneral { ...testProps } /> );
+			expect( container.querySelectorAll( '[name="blog_public"]' ).length ).toBe( 4 );
+		} );
+
+		[
+			[ 'Coming soon', '.is-coming-soon [name="blog_public"]', 1 ],
+			[ 'Public', '.is-public [name="blog_public"]', -1 ],
+			[ 'Hidden', '.is-hidden [name="blog_public"]', -1 ],
+			[ 'Private', '.is-private [name="blog_public"]', 1 ],
+		].forEach( ( [ name, selector, initialBlogPublic ] ) => {
+			test( `${ name } option should be selectable`, () => {
+				testProps.fields.blog_public = initialBlogPublic;
+				const { container, rerender } = render( <SiteSettingsFormGeneral { ...testProps } /> );
+
+				const radioButton = container.querySelector( selector );
+				expect( radioButton.checked ).toBe( false );
+				fireEvent.click( radioButton );
+				rerender( <SiteSettingsFormGeneral { ...testProps } /> );
+				expect( radioButton.checked ).toBe( true );
+			} );
+		} );
+
+		test( `Selecting Hidden should switch radio to Public`, () => {
+			testProps.fields.blog_public = -1;
+			const { container, rerender } = render( <SiteSettingsFormGeneral { ...testProps } /> );
+
+			const hiddenCheckbox = container.querySelector( '.is-hidden [name="blog_public"]' );
+			expect( hiddenCheckbox.checked ).toBe( false );
+
+			const publicRadio = container.querySelector( '.is-public [name="blog_public"]' );
+			expect( publicRadio.checked ).toBe( false );
+
+			fireEvent.click( hiddenCheckbox );
+
+			rerender( <SiteSettingsFormGeneral { ...testProps } /> );
+
+			expect( hiddenCheckbox.checked ).toBe( true );
+			expect( publicRadio.checked ).toBe( true );
+		} );
+
+		test( `Hidden checkbox should be possible to unselect`, () => {
+			testProps.fields.blog_public = -1;
+			const { container, rerender } = render( <SiteSettingsFormGeneral { ...testProps } /> );
+
+			const hiddenCheckbox = container.querySelector( '.is-hidden [name="blog_public"]' );
+			expect( hiddenCheckbox.checked ).toBe( false );
```

Comment:
Not a biggy here since checking `.checked` is pretty simple, but more of a tip/fyi.

If you `import '@testing-library/jest-dom/extend-expect';` in your test file it adds extra DOM-centric assertions. So in this case you could do:
```suggestion
			expect( hiddenCheckbox ).not.toBeChecked();
```

Like a said it's not a big deal here but it'd be resilient to things like the implementation changing to using `aria-checked="true"`. It's more useful for complicated this like `.toHaveStyle()` and `.toHaveFocus()`.